vcd/config_test.go

```diff
@@ -314,9 +323,155 @@ func TestMain(m *testing.M) {
 		}
 	}
 
+	createSuiteCatalogAndItem(testConfig)
+
 	// Runs all test functions
 	exitCode := m.Run()
 
+	destroySuiteCatalogAndItem(testConfig)
+
 	// TODO: cleanup leftovers
 	os.Exit(exitCode)
 }
+
+func createSuiteCatalogAndItem(config TestConfig) {
+	fmt.Printf("Creating resources for test suite...\n")
+	vcdClient, err := getTestVCDFromJson(config)
+	if vcdClient == nil || err != nil {
+		panic(err)
+	}
+	err = vcdClient.Authenticate(config.Provider.User, config.Provider.Password, config.Provider.SysOrg)
+	if err != nil {
+		panic(err)
+	}
+
+	org, err := govcd.GetOrgByName(vcdClient, config.VCD.Org)
+	if err != nil || org == (govcd.Org{}) {
+		panic(err)
+	}
+
+	var catalog govcd.Catalog
+
+	if testConfig.VCD.Catalog.Name == "" {
+		fmt.Printf("Creating catalog for test suite...\n")
+		catalog, err = org.CreateCatalog(testSuiteCatalogName, "Test suite purpose")
+		if err != nil || catalog == (govcd.Catalog{}) {
+			panic(err)
+		}
+		fmt.Printf("Catalog created successfuly\n")
+
+	} else {
+		fmt.Printf("Skiping catalog creation - found preconfigured one: %s \n", testConfig.VCD.Catalog.Name)
+
+		catalog, err = org.FindCatalog(testConfig.VCD.Catalog.Name)
+		if err != nil || catalog == (govcd.Catalog{}) {
+			fmt.Printf("Preconfigured catalog wasn't found \n")
+			panic(err)
+		}
+
+		fmt.Printf("Catalog found successfuly\n")
+		testSuiteCatalogName = testConfig.VCD.Catalog.Name
+	}
+
+	if testConfig.VCD.Catalog.CatalogItem == "" {
+		fmt.Printf("Creating catalog item for test suite...\n")
+		task, err := catalog.UploadOvf("../test-resources/"+config.Ova.OvaTestFileName, testSuiteCatalogOVAItem, "Test suite purpose", 20*1024*1024)
```

Comment:
The hard coded path `../test-resources` should be checked for existence. Even better, its path should be found as relative to the test file, as done in `getConfigStruct` (`config_test.go`)

Reply:
Fixed